pkg/roachprod/vm/aws/aws.go

```diff
@@ -1737,3 +1772,138 @@ func (p *Provider) ListLoadBalancers(*logger.Logger, vm.List) ([]vm.ServiceAddre
 	// This Provider has no concept of load balancers yet, return an empty list.
 	return nil, nil
 }
+
+func (p *Provider) GetVMArchitecture(l *logger.Logger, v *vm.VM) (vm.CPUArch, error) {
+
+	// If VM is tagged with arch FIPS, we use it
```

Comment:
Why? Could we infer it from the AMI (and cpu arch)?

Reply:
I considered it and went with the tag way because:
- on GCP, it requires loading all the disks (`compute disks list --filter "user:(vm1,vm2,vm3,vm4,...)`), which adds an extra 3-4s to the sync time
- on AWS, `describe-instance` only gives you the AMI ID, from which you'd have two ways to find out if it's `fips` or not:
  - use the `config.json` file:
    -> would work well if AMI ID matches the current `fips` in the hardcoded conf
    -> wouldn't match a previous build (what happens when we upgrade Ubuntu or rebuild the base image for whatever reason?)
  - run another cli command (`aws ec2 describe-images --ami-ids`) to get the name and match on name, which adds extra time (though, way less listing disks on GCP)

Change would be minimal if we want to implement it.

Reply:
Bummer. Ideally, it would be nice to remove the reliance on labels as a source of truth. Having AMI available is useful, e.g., as debugging context. Let's address this in a follow-up issue? (Something like, remove `arch` label entirely and add discovery for AMI.)

Reply:
I just finished implementing it :)

I went with the 2nd option for AWS, querying only once for all image-ids. I also refactored the code that gets the volume info to have it done in one call instead of one per VM, which will save up some time.

No real solution to speed up GCP, it will take a bit more time to sync.